packages/@internationalized/date/src/conversion.ts

```diff
@@ -187,7 +187,7 @@ export function fromAbsolute(ms: number, timeZone: string): ZonedDateTime {
   let second = date.getUTCSeconds();
   let millisecond = date.getUTCMilliseconds();
 
-  return new ZonedDateTime(year < 1 ? 'BC' : 'AD', year < 0 ? Math.abs(year) + 1 : year, month, day, timeZone, offset, hour, minute, second, millisecond);
+  return new ZonedDateTime(year < 1 ? 'BC' : 'AD', year < 1 ? -year + 1 : year, month, day, timeZone, offset, hour, minute, second, millisecond);
```

Comment:
The logic in #6542 was slightly wrong: UTC year = 0 should map to BC 1, where this would result in BC 0. This was corrected by the constraining logic in the constructor so the result ended up being right, but I updated this for clarity.